content/installation/upgrading-manager.md

```diff
@@ -2,96 +2,113 @@
-      ```curl -o ~/cfy_teardown_4_0_0.sh https://raw.githubusercontent.com/cloudify-cosmo/cloudify-dev/master/scripts/cfy_teardown_4_0_0.sh```<br>
+## Backup Agent Certificates
 
-      ```sudo bash cfy_teardown_4_0_0.sh```. You must supply an -f flag.<br>
-      For high availability clusters, you must also run ```https://github.com/cloudify-cosmo/cloudify-dev/blob/master/scripts/delete_cluster_4_0_1.py```.
+To backup agent certificates:
 
-1. (Optional) We recommend that you run this command to remove the profile directory of this Cloudify Manager from your local `~/.cloudify/profiles` directory:
-   ```rm -rf ~/.cloudify/profiles/{{ your Manager’s IP address }}```
+* Backup the `/etc/cloudify/ssl` directory to a remote host.
 
-    Cloudify Manager is now completely removed from the virtual instance.
+## Uninstall Cloudify Manager
 
-1. [Install]({{< relref "installation/installing-manager.md" >}}) a new Cloudify Manager.
+For instructions on how to uninstall the Cloudify Manager 4.3 and above, go to: [Uninstalling Cloudify Manager]({{< relref "installation/installing-manager.md#uninstalling-cloudify-manager" >}})
+
+To uninstall Cloudify Manager 4.1 or above:
+
+* To remove Cloudify Manager, run: ```cfy teardown -f```
+
+To uninstall Cloudify Manager 4.0.x:
+
+1. To download the teardown script, run: ```curl -o ~/cfy_teardown_4_0_0.sh https://raw.githubusercontent.com/cloudify-cosmo/cloudify-dev/master/scripts/cfy_teardown_4_0_0.sh```
+1. To remove Cloudify Manager, run: ```sudo bash cfy_teardown_4_0_0.sh -f```
+1. Apply ```patch-1```.
+1. (For high availability clusters only) To download the script to delete the cluster configuration, run: ```https://github.com/cloudify-cosmo/cloudify-dev/blob/master/scripts/delete_cluster_4_0_1.py```
+1. (For high availability clusters only) To run the script to delete the cluster configuration, run: ```sudo delete_cluster_4_0_1.py```
+1. (Optional) We recommend that you remove the profile directory of this Cloudify Manager from your local `~/.cloudify/profiles` directory: ```rm -rf ~/.cloudify/profiles/{{ your Manager’s IP address }}```
+
+## Install Cloudify Manager
+
+For instructions on how to install the Cloudify Manager, go to: [Installing Cloudify Manager]({{< relref "installation/installing-manager.md" >}})
+
+## Restore Snapshot
+
+To restore the snapshot:
 
 1. To upload the previously created snapshot to the new Manager, run:
-   ```cfy snapshots upload {{ /path/to/the/snapshot/file }} --snapshot-id my_snapshot```<br>
+   ```cfy snapshots upload {{ /path/to/the/snapshot/file }} --snapshot-id my_snapshot```
 
-1. To restore the snapshot to Cloudify 4.x and above, run: <br>
+1. To restore the snapshot to Cloudify 4.x and above, run:
 
   * For Cloudify Manager 4.0.1 and above: ```cfy snapshots restore my_snapshot --restore-certificates```
   * For Cloudify Manager 4.0.0: ```cfy snapshots restore my_snapshot```
 
 1. To check the status of the execution after it is complete, run:
```

Comment:
And after you do this if you're restoring from pre-4.2 and using multiple users then you need to restore the rest-security.conf.

If you're restoring a 4.2+ snapshot then the rest service will restart shortly after the execution finishes, so you should wait at least 10 seconds (new users will be usable after it restarts).

Reply:
I'll add:
(For upgrade from Cloudify Manager 4.1.x and below) Restore the rest-security.conf file to ____.
(For upgrade from Cloudify Manager 4.2 and above) When the snapshot restore is complete, the REST service restarts. (Duration > 10 secs)

Reply:
It's more involved than that- once you've restored the rest-security.conf you have to then restore the admin user from the snapshot into the database. This is probably not something that we can reliably quickly document so I suggest we leave it as a 'contact support' note. Anyone not wanting to ask for support on it is able to just reset the user passwords.

4.2+ though, yes, the wait is required.